deep_qa/layers/attention/matrix_attention.py

```diff
@@ -1,10 +1,11 @@
 from copy import deepcopy
+from typing import Dict
 
 from keras import backend as K
 from overrides import overrides
 
 from ..masked_layer import MaskedLayer
-from ...common.params import Params
+from ...common.params import pop_choice_with_default
```

Comment:
You still have the same problem though, right - you have to explicitly set `allow_default` to be True if you want to be able to use choices[0]. With two methods, the usage is implied by the name, whereas here you would(I would) definitely have to check.